src/normal.cc

```diff
@@ -2082,7 +2082,7 @@ void select_whole_buffer(Context& context, NormalParams)
 void keep_selection(Context& context, NormalParams p)
 {
     auto& selections = context.selections();
-    const int index = p.count ? p.count-1 : selections.main_index();
+    const int index = (selections.main_index() + p.count) % selections.size();
```

Comment:
What's slightly surprising here is that `1<space>` behaves differently than `<space>`.
For example on

```
%(foo)
%(bar)
```

where the first selection is the main one.

I'm not sure what's the best behavior here.

Reply:
The default count in Kakoune is 0, so I think this behaviour makes sense and is predictable.

Reply:
But from the user perspective all commands default to 1, right? At least I couldn't find one that doesn't.

Reply:
You could say that `s`,`<a-k>` and their variants are 0-based, since the default behavior is to use the 0 capture group. I think it's fine that different keys have a different default, depending on what makes sense.

Reply:
Correct, the default count value is not consistent across all primitives. C.f. #2306